src/Modules/OrchardCore.Commerce.Payment/Controllers/PaymentController.cs

```diff
@@ -128,11 +128,15 @@ await _paymentProviders
         }
         catch (Exception exception)
         {
-            _logger.LogError(exception, "An exception has occurred during checkout form validation.");
-
+            string error = shoppingCartId == null ?
+                   $"An exception has occurred during checkout form validation."
+                 : $"An exception has occurred during checkout form validation at {shoppingCartId}.";
+#pragma warning disable CA2254
+            _logger.LogError(exception, message: error);
+#pragma warning restore CA2254
             var errorMessage = HttpContext.IsDevelopmentAndLocalhost()
                 ? exception.ToString()
-                : T["An exception has occurred during checkout form validation."].Value;
+                : T[error].Value;
```

Comment:
Never put variables with string interpolation in `T[...]` because it will never work (you'd have to localize all possible values). A little text repetition is not the end of the word, and I think this is more readable anyway:

```csharp
            var shoppingCartIdForDisplay = shoppingCartId == null ? "(null)" : $"\"{shoppingCartId}\"";

            _logger.LogError(
                exception,
                "An exception has occurred during checkout form validation for shopping cart ID {ShoppingCartId}.",
                shoppingCartIdForDisplay);
            var errorMessage = HttpContext.IsDevelopmentAndLocalhost()
                ? exception.ToString()
                : T["An exception has occurred during checkout form validation for shopping cart ID {0}.", shoppingCartIdForDisplay].Value;
```
(GitHub doesn't let me make this a "suggestion", but please overwrite the displayed code with this snippet.)